### ISSUE TITLE
hangs at Directory Search

### ISSUE THREAD

**Issue by mvsite** [yogeshojha/rengine#819]
### Issue Summary

reNgine is stuck on the Directory Search step and does not recover.

### Steps to Reproduce

1. I installed the current reNgine as of 20230217 using git clone, correct the bugs preventing it from working (updating versions of go and celery in the right files - see other bug reports for details) and then connected to it successfully.
2. I added a domain, chose Full Scan setting, and started the scan.
3. reNgine successfully completes each step until Directory Search then it just hangs and always shows Directory Search as "In progress".
4. I looked in the docker containers for running processes that could give a clue and saw a defunct fuff that I couldn't kill -9, so suspect this is where the problem is. There was no parent process listed for fuff.  I also tried restarting the docker container (surely not helpful).  Apparently reNgine doesn't care about it's scan state and didn't pick up on that something in progress wasn't completed. 

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

Yes this is broken, it should either work, or failing to work should have **timed out** and killed the process and moved on (reporting the issue in the db and interface somewhere). To me, that is the whole point of this software is to manage the various tools responsibly (meaning, to prevent issues like this where there is some zombie process) and gather what data we can in this imperfect sometimes unlucky world.

* I have confirmed that this issue can be reproduced as described on a latest version/pull of reNgine: (yes / no)

yes

### Technical details

Please list out any technical details such as operating environment.
Ubuntu 22.04, having followed the rengine wiki install document meticulously as per requirements, and using latest docker that is installer when following the instructions. (no errors during these steps except deprecation warning about gpg etc).

Here is some info about the state of the containers, unchanged in nearly 10 hours#818 

```
# docker ps -q | xargs -I {} docker exec {} ps aux
PID   USER     TIME  COMMAND
    1 tor       0:58 /usr/bin/tor -f /etc/tor/torrc
 8342 tor       0:00 ps aux
PID   USER     TIME  COMMAND
    1 root      0:00 nginx: master process nginx -g daemon off;
   29 nginx     0:00 nginx: worker process
   30 nginx     0:01 nginx: worker process
   31 nginx     0:00 nginx: worker process
   32 nginx     0:00 nginx: worker process
   44 root      0:00 ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0   3984  3028 ?        Ss   Feb17   0:00 /bin/bash /usr/src/app/entrypoint.sh
root          68  0.0  0.7  74588 57832 ?        S    Feb17   0:00 python3 manage.py runserver 0.0.0.0:8000
root         578  3.8  2.0 956540 165964 ?       Rl   Feb17  22:10 /usr/bin/python3 manage.py runserver 0.0.0.0:8000
root         784  0.0  0.0   5900  2844 ?        Rs   03:03   0:00 ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  1.6 175296 135156 ?       Ss   Feb17   0:35 /usr/bin/python3 /usr/local/bin/celery -A reNgine beat -l INFO --scheduler django_celery_beat.schedulers:DatabaseScheduler
root         193  0.0  0.0   5900  2932 ?        Rs   03:03   0:00 ps aux
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.1  0.1 312272 14228 ?        Ssl  Feb17   1:05 /usr/bin/python3 /usr/local/bin/watchmedo auto-restart --recursive --pattern=*.py --directory=/usr/src/app/reNgine/ -- celery -A reNgine worker --autoscale=30,5 -l INFO
root         388  0.2  0.0      0     0 ?        Z    Feb17   1:32 [celery] <defunct>
root        2169  0.0  0.0      0     0 ?        Z    Feb17   0:00 [RDD Process] <defunct>
root        2232  0.0  0.0      0     0 ?        Z    Feb17   0:00 [Utility Process] <defunct>
root        2384  0.0  0.0      0     0 ?        Z    Feb17   0:00 [RDD Process] <defunct>
root        2385  0.0  0.0      0     0 ?        Z    Feb17   0:00 [Utility Process] <defunct>
root        2471  0.0  0.0      0     0 ?        Z    Feb17   0:00 [RDD Process] <defunct>
root        2488  0.0  0.0      0     0 ?        Z    Feb17   0:00 [Utility Process] <defunct>
root        3139  0.0  0.0      0     0 ?        Z    Feb17   0:03 [ffuf] <defunct>
root        3167  0.1  1.7 183724 142644 ?       Ss   Feb17   1:09 /usr/bin/python3 /usr/local/bin/celery -A reNgine worker --autoscale=30,5 -l INFO
root        3195  0.0  1.4 182616 120884 ?       S    Feb17   0:00 /usr/bin/python3 /usr/local/bin/celery -A reNgine worker --autoscale=30,5 -l INFO
root        3196  0.0  1.4 182620 120884 ?       S    Feb17   0:00 /usr/bin/python3 /usr/local/bin/celery -A reNgine worker --autoscale=30,5 -l INFO
root        3197  0.0  1.4 182624 120884 ?       S    Feb17   0:00 /usr/bin/python3 /usr/local/bin/celery -A reNgine worker --autoscale=30,5 -l INFO
root        3198  0.0  1.4 182628 120884 ?       S    Feb17   0:00 /usr/bin/python3 /usr/local/bin/celery -A reNgine worker --autoscale=30,5 -l INFO
root        3199  0.0  1.4 182888 120888 ?       S    Feb17   0:00 /usr/bin/python3 /usr/local/bin/celery -A reNgine worker --autoscale=30,5 -l INFO
root        3254  0.0  0.0   5900  2788 ?        Rs   03:03   0:00 ps aux
PID   USER     TIME  COMMAND
    1 postgres  0:06 postgres
   47 postgres  0:00 postgres: checkpointer
   48 postgres  0:01 postgres: background writer
   49 postgres  0:01 postgres: walwriter
   50 postgres  0:01 postgres: autovacuum launcher
   51 postgres  0:06 postgres: stats collector
   52 postgres  0:00 postgres: logical replication launcher
 1364 postgres  0:00 postgres: rengine rengine 172.19.0.6(58394) idle
 9581 postgres  0:00 postgres: rengine rengine 172.19.0.5(56758) idle
 9582 root      0:00 ps aux
PID   USER     TIME  COMMAND
    1 redis     1:41 redis-server *:6379
   26 root      0:00 ps aux

```

**Comment by mvsite** [comment]
btw I fixed this by upgrading ffuf to 2.0.0 (manually in docker since autoupdate failed - using github.com/ffuf/ffuf/v2@latest) and also by setting a timeout in the yaml for full scan in "dir_file_fuzz:" section. I'm not sure which was the key fix and don't care enough to undo the change to find out.  No one else seems to be complaining about this bug so maybe it was just something specific I encountered.
